### PR TITLE
Use local config file in dev. Fix #903

### LINKED ISSUE #903: Use a local .hyper.js config file and and .hyper_plugins directory in DEV
It would be great to optionally have a config file and a plugin directory in root project directory.
It will offer the possibility to develop features or plugins, and continue to use a Hyper.app stable version for daily tasks.

### PR DESCRIPTION
**Pull request by chabou**

### CODE REVIEW

File: app/plugins.js [hunk 1]
```diff
@@ -16,8 +16,8 @@ const notify = require('./notify');
 const cache = new Config();
 
 // modules path
-const path = resolve(homedir(), '.hyper_plugins');
-const localPath = resolve(homedir(), '.hyper_plugins', 'local');
+const path = resolve(config.getConfigDir(), '.hyper_plugins');
```
Comment (albinekb) [inline app/plugins.js]:
```js
const path = resolve(config.getConfigDir(), '.hyper_plugins')
const localPath = resolve(path, 'local')
```
?

Comment (chabou) [inline app/plugins.js]: oups

File: .gitignore [hunk 2]
```diff
@@ -6,3 +6,8 @@ node_modules
 
 # logs
 npm-debug.log
+
+# optionnal dev config file and plugins directory
```
Comment (TooTallNate) [inline .gitignore]: nit: `s/optionnal/optional/`

File: app/plugins.js (hunk 1 above)
Comment (chabou) [inline app/plugins.js]: yes better :)

File: .gitignore (hunk 2 above)
Comment (chabou) [inline .gitignore]: ty :) it is "optionnel" in french

**Comment by chabou** [comment]
Do you plan to merge it ? 😊 
Really cool to have a no-plugin config to work on Hyper app or a special config to work on a local and experimental plugin. (And continue to use prod version of Hyper in the same time).
I can add instructions in `Contribute` section of Readme but warning : I'm not an english native speaker 😆

**Comment by ppot** [comment]
@chabou first this need rebase.

**Comment by chabou** [comment]
What's the hell...
I made rebase periodically but my last one (yesterday) doesn't appear here. This PR is not synchronized with my branch anymore : https://github.com/chabou/hyper/tree/dev_use_local_config

I will check that

**Comment by chabou** [comment]
@ppot I made a rebase 8 hours ago and this PR seems up-to-date, no ?

**Comment by ppot** [comment]
@chabou I can't see when the rebase is done 😂

**Comment by chabou** [comment]
Ahah of course but I made a rebase before my comment "Do you plan..."
So I didn't understand your comment 😄

**Comment by ppot** [comment]
This will need some upgrades. But okay for a start!

**Pull request merged**